File: java8/src/main/java/folio/codinginterview/domain/StockSymbol.java
package folio.codinginterview.domain;

import java.util.Optional;

/** 銘柄を表す。 */
public enum StockSymbol {
    Toyopa, Somy;

    public static Optional<StockSymbol> fromString(String s) {
        if ("Toyopa".equals(s)) return Optional.of(Toyopa);
        if ("Somy".equals(s)) return Optional.of(Somy);
        return Optional.empty();
    }
}

File: typescript/src/domain/stockSymbol.ts
/** 銘柄を表す。 */
export type StockSymbol = "Toyopa" | "Somy";

export const StockSymbol = {
  Toyopa: "Toyopa" as StockSymbol,
  Somy: "Somy" as StockSymbol,

  fromString(s: string): StockSymbol | undefined {
    if (s === "Toyopa" || s === "Somy") return s;
    return undefined;
  },
};
